### PR TITLE
Fix Issue 556

### PR DESCRIPTION
**Pull request by diegomurakami**
## Description
Fix https://github.com/adazzle/react-data-grid/issues/556

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
New checkFocus method was implemented recently on Cell component. This method checks the selected cell when filters are selected and used. The problem is that [0,0] is selected when filters are used, causing [0,0] to be focused after filter change.

**What is the new behavior?**
When the user clicks on the Header, we call a new method onHeaderClick which sets [-1,-1] as the current selection of the Grid using onCellClick from cellMetaData (injected as a new prop to the Header component).

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/Header.js [hunk 1]
```diff
@@ -179,6 +181,11 @@ const Header = React.createClass({
     return createObjectWithProperties(this.props, knownDivPropertyKeys);
   },
 
+  // Set the cell selection to -1 x -1 when clicking on the header
+  onHeaderClick() {
```
Comment (malonecj) [inline src/Header.js]: @diegomurakami please add a test for this

**Comment by kbergin** [comment]
@diegomurakami Thank you for fixing this! Looking forward to the merge.

File: src/Header.js (hunk 1 above)
Comment (diegomurakami) [inline src/Header.js]: @malonecj sorry for the delay. It's done.

File: src/__tests__/Header.spec.js [hunk 2]
```diff
@@ -178,5 +187,13 @@ describe('Header Unit Tests', () => {
...
       expect(headerDiv.props().getValidFilterValues).toBeUndefined();
     });
+
+    it('execute onCellClick event on cellMetaData', () => {
+      spyOn(testAllProps.cellMetaData, 'onCellClick');
+      const wrapper = renderComponent(testAllProps);
+      const headerDiv = wrapper.find('div');
+      headerDiv.simulate('click');
+      expect(testAllProps.cellMetaData.onCellClick).toHaveBeenCalled();
+    });
```
Comment (malonecj) [inline src/__tests__/Header.spec.js]: You should be verifying the parameters of cellMetaData. Its important to verify that the selected idx and rowIdx is -1,-1

**Comment by diegomurakami** [comment]
@malonecj haha thanks !

**Pull request merged**